Plan/api/src/main/java/com/djrapitops/plan/component/ComponentService.java

```diff
@@ -0,0 +1,177 @@
+/*
+ *  This file is part of Player Analytics (Plan).
+ *
+ *  Plan is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License v3 as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Plan is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Plan. If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.djrapitops.plan.component;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A utility api class for dealing with rich and less-rich Minecraft message formats.
+ *
+ * @author Vankka
+ */
+public interface ComponentService {
```

Comment:
I don't know if the intention was that extension code would call the service directly to translate values before they hit the database - but I think it is possible to do the translation for them on the Plan side, possibly after the data is fetched from database. (Both tables and StringProvider support formatting after the query)

If the translation is implemented that way, it is not necessary to expose a new API for extensions.

I would suggest against exposing this service as an API in its current form, since it is unclear how it is to be used in relation to the existing apis

Reply:
I was thinking there could be a `ComponentProvider` or similar, and you would give it a `Component`, eg.

```java
@ComponentProvider(...)
public Component displayName(UUID playerUUID) {
    String displayName = magic.getDisplayNameForUUID(playerUUID);
    return ComponentService.getInstance().fromLegacy(displayName)
}
```
The main benefit of doing this would be to allow the extension maker to specify the input format, instead of just relying on auto determine all the time.

Although it seems kind of unlikely there might be a case where content should not be formatted but contains a legacy code (eg. `&` + a-f)

I do agree that exposing the `to***` methods is confusing, and there doesn't seem to be a good use for them outside of Plan's own code.

Reply:
Alright that makes sense 🙂